### PR TITLE
[Import Wizard 2] Add server-side column detection to the mapping page

### PR DESCRIPTION
**Pull request by juliawu**
This PR:
1) Adds column detection to the mapping page via the server-side API
2) Moves client-side detection files from `import_wizard/` to `import_wizard2/`

This does NOT add template as an extra parameter to the API, slating that for a separate commit.

This is my first time writing TypeScript, so extra scrutiny on the implementation would be appreciated!

### CODE REVIEW

File: static/js/import_wizard2/components/mapping_page.tsx [hunk 1]
```diff
@@ -55,9 +56,32 @@ export function MappingPage(props: MappingPageProps): JSX.Element {
   }
 
   useEffect(() => {
-    // TODO(beets): Use server-side detection API.
-    const predictedMapping = getPredictions(props.csvData, placeDetector);
-    setPredictedMapping(predictedMapping);
+    /* Build JSON of column data for detection API Request */
+    const column_ids = new Map(
```
Comment (chejennifer) [inline static/js/import_wizard2/components/mapping_page.tsx:60]: we want to use camelcase for variables in typescript.

File: static/js/import_wizard2/components/mapping_page.tsx [hunk 2]
```diff
@@ -55,9 +56,32 @@ export function MappingPage(props: MappingPageProps): JSX.Element {
...
+      })
+    );
+    const columnParameters = new Map<string, unknown>([
+      ["column_ids", column_ids],
+      ["column_headers", column_headers],
+      ["column_values", props.csvData.columnValuesSampled],
+    ]);
+    /* Use server-side API for column detection */
+    axios
+      .post(`/api/detection/detect_columns`, JSON.stringify(columnParameters))
```
Comment (chejennifer) [inline static/js/import_wizard2/components/mapping_page.tsx:77]: have you checked that JSON.stringify(columnParameters) will give you the correct string? I remember running into problems with JSON.stringify on the Map type in the past. If Map type doesn't work, I would suggest just using an object for everything

```
const columnIds = {};
props.csvData.orderedColumns.forEach((column) => {
columnIds[column.columnIdx] = column.id;
});
...
const columnParameters = {
column_ids: columnIds,
...
}
```

File: static/js/import_wizard2/components/mapping_page.tsx [hunk 3]
```diff
@@ -55,9 +56,32 @@ export function MappingPage(props: MappingPageProps): JSX.Element {
...
+      props.csvData.orderedColumns.map((column) => {
+        return [column.columnIdx, column.id];
+      })
+    );
+    const column_headers = new Map(
+      props.csvData.orderedColumns.map((column) => {
+        return [column.columnIdx, column.header];
+      })
+    );
+    const columnParameters = new Map<string, unknown>([
```
Comment (chejennifer) [inline static/js/import_wizard2/components/mapping_page.tsx:70]: usually would prefer not to use the "unknown" type

File: static/js/import_wizard2/components/mapping_page.tsx [hunk 4]
```diff
@@ -55,9 +56,32 @@ export function MappingPage(props: MappingPageProps): JSX.Element {
...
+    ]);
+    /* Use server-side API for column detection */
+    axios
+      .post(`/api/detection/detect_columns`, JSON.stringify(columnParameters))
+      .then((resp) => {
+        const parsedResponse = parseDetectionApiResponse(resp.data);
+        setPredictedMapping(parsedResponse);
+      })
+      .catch((error) => {
+        console.log(error);
```
Comment (chejennifer) [inline static/js/import_wizard2/components/mapping_page.tsx:83]: maybe set predictedMapping to an empty mapping (i.e., `new Map()`)

**Review by chejennifer — COMMENTED** [review]
nice job, this looks good!

File: static/js/import_wizard2/components/mapping_page.tsx (hunk 1 above)
Comment (beets) [reply static/js/import_wizard2/components/mapping_page.tsx:60]: let's also pull this out into a helper function so that we can easily add tests

Comment (juliawu) [reply static/js/import_wizard2/components/mapping_page.tsx]: Moved to a helper function!

File: static/js/import_wizard2/components/mapping_page.tsx (hunk 2 above)
Comment (juliawu) [reply static/js/import_wizard2/components/mapping_page.tsx]: Replaced Map with Object

File: static/js/import_wizard2/components/mapping_page.tsx (hunk 3 above)
Comment (juliawu) [reply static/js/import_wizard2/components/mapping_page.tsx]: Switching to object fixed this!

File: static/js/import_wizard2/components/mapping_page.tsx (hunk 4 above)
Comment (juliawu) [reply static/js/import_wizard2/components/mapping_page.tsx:91]: added!

**Review by chejennifer — COMMENTED** [review]
thanks for the changes! few small nits

**Pull request closed without merging**